static/ast-view.js

```diff
@@ -37,7 +40,7 @@ function Ast(hub, container, state) {
     this.astEditor = monaco.editor.create(this.domRoot.find(".monaco-placeholder")[0], {
         value: "",
         scrollBeyondLastLine: false,
-        language: 'cppp', //we only support cpp for now
+        language: languages[state.lang] ? languages[state.lang].monaco : 'cppp',
```

Comment:
At this point, `state.lang` is not defined, and so this always sets the language to `cppp`. How to get the language at this point? Perhaps from `state.id` or `state.editorid`, but how?

Reply:
Why do you want to change the language of the editor in the first place? This language takes care mostly of syntax highlighting, which I guess is not what you want.

As for why it's not working, it's because the ast pane does not recive a langId on its state upon creation. Only those panes who need it (Editor and compiler) do.

If you really need to change the language here for some reason I can't see, let me know and we can work out how to proceed

Reply:
The astEditor is not an editor, it's the AST view. D's "AST" output is just D code, so the language should be set to 'd' to get correct highlighting.  (If I manually just set it to `'d'`, it all works and I'm happy ;))

Reply:
Oh, in this case, I think we can cheat a bit here, instead of having to rely on adding a setting for every language. I'll show you what I mean, sec

Reply:
Cool, note that the _exact_ same issue is present in the opt view ("Clang optimization output").

Reply:
I'm on it